### PR TITLE
update api proto to include new fields

### PR DESCRIPTION
**Pull request by zachpuck**
Updated Api with new fields. Unclear if we want to add machine SSH details to the GetCluster reply.
Also unsure if we should add machineSet now and if we want to make instanceType an enum.

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -205,26 +215,27 @@ message KubernetesLabel {
...
-    // key they can still impersonate the machine. TODO: implement known_hosts.
-    string password = 4;
     // The labels for the machines
     repeated KubernetesLabel labels = 5;
-    // optional public host ip
-    string publichost = 6;
+    // The list of kubernetes taints
+    repeated KubernetesTaint taints = 7;
+    // Type of machine to provision (small, medium, gpu, ...)
+    string instanceType = 8; // TODO: (zachpuck) Do we want instance type as enum?
```
Comment (oneilcin) [inline api/api.proto]: We have not decided to pick strict instanceType names yet, so leave as a string.

File: api/api.proto [hunk 2]
```diff
@@ -205,26 +215,27 @@ message KubernetesLabel {
...
+    // The value of a taint
+    string value = 2;
+    // The effect of a taint
+    KubernetesTaintEffect effect = 3;
+}
+
 // The specification for a specific node
 message SshMachineSpec {
     // The username for SSH access
     string username = 1;
```
Comment (oneilcin) [inline api/api.proto]: I think we can get rid of username, unless we are planning to support both ssh and Maas?

File: api/api.proto [hunk 3]
```diff
@@ -205,26 +215,27 @@ message KubernetesLabel {
     string value = 2;
 }
 
+message KubernetesTaint {
+    // The name of a taint
```
Comment (davidewatson) [inline api/api.proto]: Comments should begin with the element they describe for the purposes of godoc. Cf.

> Notice this comment is a complete sentence that begins with the name of the element it describes. - https://blog.golang.org/godoc-documenting-go-code

Ergo, 
```// Foo is an example function
func Foo() {
``` 
and not
```// An example function
func Foo() {
```

File: api/api.proto (hunk 2 above)
Comment (davidewatson) [reply api/api.proto]: I don't think we need SSH access, though I can imagine it being useful later. Maybe don't remove `username` now, or if you do, also rename this message/structure...

File: api/api.proto (hunk 3 above)
Comment (zachpuck) [reply api/api.proto]: It looks like almost all of the comments in the api.proto file are not in this format. Changing this comment to meet Go commenting standards will likely mean changing all these comments. Deferring to a separate effort for comment standardizing.

File: api/api.proto (hunk 2 above)
Comment (zachpuck) [reply api/api.proto]: leaving username for now. If we determine it is never needed we can remove at that later date.

**Pull request merged**